doc/designs/passkeys.md

```diff
@@ -0,0 +1,246 @@
+# Passkey authentication
+
+## Overview
+
+Traditional authentication with a password is not considered secure enough
+by many companies or government agencies. Alternate and more secure
+solutions exist, among which the use of passkeys, where the private
+key is stored on the device and the server only needs to know the public
+key.
+
+The goal of this feature is to use a passkey to authenticate a user
+against IPA.
+
+The project will be jointly developed by SSSD and IPA:
+
+- IPA provides the interface to store the user's public credentials
+- IPA provides the interface to configure passkey settings
+- SSSD performs the actual authentication
+
+## Use Cases
+
+- The administrator or the user registers a passkey into IPA, associated
+to a user account. The registration requires the insertion of the key, or
+the string output provided by SSSD registration mechanism.
+- The user is then able to authenticate to any IPA enrolled host using the
+passkey. Authentication methods include console login, desktop login,
+or su access.
+
+## How to Use
+
+### Configuration of the passkey settings by the administrator
+
+The administrator is able to specify common settings that will apply:
+
+- require user verification during authentication (On/Off/Default):
+  - On: require user verification during authentication (PIN for instance).
+  - Off: do not require user verification during authentication.
+  - Default: fallback to the passkey’s default behavior.
+
+### Registration of credentials
+
+The user can register credentials for himself, or the admin (or any user with
+the permission "System: Manage User passkeys") can register
+credentials for another user.
+
+During the registration process, it is possible to specify
+- a COSE type: `es256`, `rs256` or `eddsa`
+- request user verification: true or false
+the authentication will force to execute the user verification check even if
+the passkey settings do not set this flag. If credentials are registered without
+the flag, the global passkey settings apply.
+
```

Comment:
Please add a sentence that explains how the credential is built and used: "When the passkey credential is registered, a relaying party (RP) is set to be the IPA domain (e.g. ipa.test). While using a domain-wide relaying party reduces access control capabilities for individual application's use of the registered passkey, IPA provides own access control mechanisms to be layered on top. We choose to combine existing authorization features of IPA with an ease of use for the passkeys."